### ISSUE TITLE
Landmark role of the modal container?

### ISSUE THREAD

**Issue by tomi-sjoblom** [scottaohara/accessible_modal_window#21]
Hey,

We came across accessibility validator errors due to the modal dialog being in 'non-landmarked area' of the page (being the first element of body). Apparently `role="dialog"` isn't a ARIA landmark in itself.

In our case using `<article>` as per https://handbook.floeproject.org/ModalDialogs.html suited the purpose well but do you think there are any side effects doing that?

**Comment by scottaohara** [comment]
The dialog, which should not be accessible to people nor exposed to the accessibility tree when not invoked, shouldn't trigger such a validation error.  When invoked, all other content of the DOM should be hidden from the accessibility tree, and only the modal dialog displayed.  

There is no actual accessibility failure if this has been implemented as described, so such a validation error is either being overly aggressive, and may indicative of other opinionated issues with the validator's rule set.  

For example, using an `article` element as the dialog wrapper should make no difference. If done as indicated in the linked post (which please know contains some inaccuracies in terminology and descriptions of how certain attributes work) the `role=dialog` is added to the `article`. This means the `article` is no longer an `article` anymore, but rather is exposed as a `dialog`... which would be the same as this script.  Additionally, an `article` is **not** a landmark, so for the validator to pass this again indicates there may be something off about the ruleset of the validator in this regard.

tldr; Changing the element to a `article` to meet the validation error (again assuming you're doing `article role=dialog`) does nothing to make the experience better or worse for people.  The only thing it does is pass a bogus validation error. :)
